framework/src/actions/CreateProblemAction.C

```diff
@@ -96,6 +100,9 @@ CreateProblemAction::act()
     _problem->useFECache(_fe_cache);
     _problem->setKernelCoverageCheck(getParam<bool>("kernel_coverage_check"));
     _problem->setMaterialCoverageCheck(getParam<bool>("material_coverage_check"));
+    // Silence Parallel Barriers
+    if (_pars.get<bool>("silence_parallel_barriers"))
+      _problem->silenceParallelBarriers();
```

Comment:
Then, here just pass the parameter into `setSilence...` like we do for the coverage checks above.